src/core/constraints/ShapeBasedConstraint.hpp

```diff
@@ -41,7 +41,7 @@ class ShapeBasedConstraint : public Constraint {
 
   Shapes::Shape const &shape() const { return *m_shape; }
 
-  ReflectionType const &reflection_type() const;
+  int &reflection_type(){return m_reflection_type;}
```

Comment:
Please leave the type as it is, and convert the parameters at the python/si level from a string parameter to the corresponding enum val.